File: PlateformRH/src/main/java/com/PlateformRH/Formation/formation.java
package com.PlateformRH.Formation;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Getter
@Setter
public class formation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titre;
    private String description;

    private Date dateDebut;
    private Date dateFin;

    @Enumerated(EnumType.STRING)
    private TypeFormation typeFormation;

    private int capacite;
}

File: rh/src/app/models/formation.ts
export interface Formation {

  id?: number;

  titre: string;

  description: string;

  dateDebut: string;

  dateFin: string;

  typeFormation: string;

  capacite: number;
}

export interface DemandeFormation {

  id?: number;

  employeId: number;

  formationId: number;

  statut: 'EN_ATTENTE' | 'APPROUVE' | 'REFUSE';
}
